src/base/utils/string.cpp

```diff
@@ -76,23 +76,23 @@ namespace
             // See #5238 and #5240
             // Without ICU library, QCollator doesn't support `setNumericMode(true)` on OS older than Win7
             // if (QSysInfo::windowsVersion() < QSysInfo::WV_WINDOWS7)
-                return lessThan(left, right);
+                return compare(left, right);
 #endif
-            return (m_collator.compare(left, right) < 0);
+            return m_collator.compare(left, right);
 #else
-            return lessThan(left, right);
+            return compare(left, right);
 #endif
         }
 
-        bool lessThan(const QString &left, const QString &right) const
+        int compare(const QString &left, const QString &right) const
         {
-            // Return value `false` indicates `right` should go before `left`, otherwise, after
+            // Return value `-1` indicates left less than right, `0` indicates equal, `1` indicates left greater than right
             int posL = 0;
             int posR = 0;
             while (true) {
                 while (true) {
                     if ((posL == left.size()) || (posR == right.size()))
-                        return (left.size() < right.size());  // when a shorter string is another string's prefix, shorter string place before longer string
+                        return (left.size() < right.size()) ? -1 : 1;  // when a shorter string is another string's prefix, shorter string place before longer string
```

Comment:
D'oh! For some reason it totally slipped my mind that they're both always positive values ... so no chance of underflow.

Without casting subtraction should definitely be more efficient.